pkg/apiclient/client_factory.go

```diff
@@ -64,11 +63,29 @@ type Client struct {
 	// May be nil if we haven't done space lookup yet
 	ActiveSpace *spaces.Space
 
-	// Handle out to prompt the user for things. If this is nil, it means we're in no-prompt mode
-	Ask question.Asker
+	Ask question.AskProvider
 }
 
-func NewClientFactory(httpClient *http.Client, host string, apiKey string, spaceNameOrID string, asker question.Asker) (ClientFactory, error) {
+func requireArguments(items map[string]any) error {
+	for k, v := range items {
+		if v == nil {
+			return fmt.Errorf("required argument %s was nil", k)
+		}
+
+		if vstr, ok := v.(string); ok && vstr == "" {
+			return fmt.Errorf("required string argument %s was empty", k)
+		}
+	}
+	return nil
+}
+
+func NewClientFactory(httpClient *http.Client, host string, apiKey string, spaceNameOrID string, ask question.AskProvider) (ClientFactory, error) {
+	if err := requireArguments(map[string]any{
```

Comment:
This doesn't seem particularly "go-ish" but google searching kept sending me down the path of using big validation libraries that use reflection etc... which seemed way to heavy just to assert that some parameters aren't nil. Open to suggestions here on how we might more easily validate non-nil parameters

Reply:
Yes, there are validation libraries to consider. I don't think we should use them in this instance. Go is very idiomatic when it comes to input validation for functions. Everyone writes the same checks:

```go
if inputParameter == nil {
  return fmt.Errorf("...", ...)
}
```

Reply:
I thought about that, but then a run of 6 `if x == nil`'s seemed like it'd be a bit long.